### PR TITLE
Support deployment in cluster mode

### PR DESCRIPTION
**Pull request by ButterBright**

### CODE REVIEW

**Review by hanahmily — CHANGES_REQUESTED** [review]

**Comment by wu-sheng** [comment]
How is this PR going?

**Comment by ButterBright** [comment]
It is still in progress. I've been a bit busy lately, but I will complete it as soon as possible.

**Comment by wu-sheng** [comment]
Could you add a two nodes BanyanDB cluster to verify your helm?

**Comment by ButterBright** [comment]
Okay.

File: README.md [hunk 1]
```diff
@@ -118,6 +118,31 @@ $ helm install my-release banyandb -f values.yaml
...
+  renewBefore: 4320h
+  issuerRef:
+    name: banyandb-issuer
+    kind: Issuer
+  usages:
+    - server auth
+    - client auth
+  commonName: banyandb-etcd
+  dnsNames:
+    - "*.banyandb-etcd.default.svc.cluster.local"
```
Comment (hanahmily) [inline README.md:142]: Please add a section explaining the reason for setting this specific URL.

File: README.md [hunk 2]
```diff
@@ -118,6 +118,31 @@ $ helm install my-release banyandb -f values.yaml
 ## Use external certificate authorities for TLS
 If you'd like to use external certificate authorities, such as Vault, corresponding annotations can be injected into [banyandb](./chart/templates/statefulset.yaml).
 
+## Setup certificate for etcd TLS
+To establish secure communication for etcd, you can leverage cert-manager to generate the necessary TLS certificates. This tool simplifies the process of creating and managing certificates.
```
Comment (hanahmily) [inline README.md:122]: Can you please add a section that clearly outlines the steps involved in the installation of cert manager?

File: README.md (hunk 1 above)
Comment (hanahmily) [reply README.md:142]: When the embed etcd is enabled, liaison and data nodes use headless services to access etcd. What is the common service used for?

File: chart/templates/statefulset.yaml [hunk 3]
```diff
@@ -228,4 +229,291 @@ spec:
...
+            - name: BYDB_ETCD_ENDPOINTS
+              value: "{{- .Values.cluster.etcdEndpoints | join "," -}}"
+            {{- else }}
+            {{- $endpoints := list }}
+            {{- $replicaCount := int .Values.etcd.replicaCount }}
+            {{- $releaseName := .Release.Name }}
+            {{- $namespace := .Release.Namespace }}
+            {{- range $i := until $replicaCount }}
+              {{- $endpoint := printf "%s-etcd-%d.%s-etcd-headless.%s:2379" $releaseName $i $releaseName $namespace }}
+              {{- $endpoints = append $endpoints $endpoint }}
```
Comment (hanahmily) [inline chart/templates/statefulset.yaml:318]: You should move this piece to `_helpers.tpl` to reuse it for deployment.

**Review by hanahmily — CHANGES_REQUESTED** [review]

**Review by hanahmily — APPROVED** [review]
LGTM

**Pull request merged**